code/modules/reagents/Chemistry-Grinder.dm

```diff
@@ -31,161 +32,138 @@
 
 	if(!istype(O))
-		return
+		return FALSE
 
-	if(holdingitems && holdingitems.len >= limit)
+	if(LAZYLEN(holdingitems) >= limit)
 		to_chat(user, SPAN_NOTICE("\The [src] cannot hold any additional items."))
-		return 1
+		return TRUE
 
 	if (istype(O,/obj/item/chems/glass) || \
 		istype(O,/obj/item/chems/food/drinks/glass2) || \
 		istype(O,/obj/item/chems/food/drinks/shaker))
 
 		if (beaker)
-			return 1
+			return TRUE
 		else
 			if(!user.unEquip(O, src))
-				return
-			src.beaker =  O
+				return FALSE
+			beaker =  O
 			update_icon()
-			src.updateUsrDialog()
-			return 0
+			SSnano.update_uis(src)
+			return FALSE
 
 	if(is_type_in_list(O, blacklisted_types))
 		to_chat(user, SPAN_NOTICE("\The [src] cannot grind \the [O]."))
-		return
+		return FALSE
 
 	if(is_type_in_list(O, bag_whitelist))
 		var/obj/item/storage/bag = O
-		var/failed = 1
+		var/failed = TRUE
 		for(var/obj/item/G in O)
 			if(!G.reagents || !G.reagents.total_volume)
 				continue
-			failed = 0
-			bag.remove_from_storage(G, src, 1)
+			failed = FALSE
+			bag.remove_from_storage(G, src)
 			holdingitems += G
-			if(holdingitems && holdingitems.len >= limit)
+			if(LAZYLEN(holdingitems) >= limit)
 				break
 
 		if(failed)
 			to_chat(user, SPAN_NOTICE("Nothing in \the [O] is usable."))
-			return 1
+			return TRUE
 		bag.finish_bulk_removal()
 
-		if(!O.contents.len)
+		if(!LAZYLEN(O.contents))
 			to_chat(user, "You empty \the [O] into \the [src].")
 		else
 			to_chat(user, "You fill \the [src] from \the [O].")
 
-		src.updateUsrDialog()
-		return 0
+		SSnano.update_uis(src)
+		return FALSE
 
 	if(O.w_class > item_size_limit)
 		to_chat(user, SPAN_NOTICE("\The [src] cannot fit \the [O]."))
 		return
 
 	if(istype(O,/obj/item/stack/material))
 		var/obj/item/stack/material/stack = O
-		var/material/material = stack.material
-		if(!length(material.chem_products))
+		var/material/material = stack.get_material()
+		if(!LAZYLEN(material.chem_products))
 			to_chat(user, SPAN_NOTICE("\The [material.display_name] is unable to produce any usable reagents."))
-			return 1
+			return TRUE
 
-	else if(!O.reagents || !O.reagents.total_volume)
+	else if(!O.reagents?.total_volume)
 		to_chat(user, SPAN_NOTICE("\The [O] is not suitable for grinding."))
-		return 1
+		return TRUE
 
 	if(!user.unEquip(O, src))
-		return
+		return FALSE
 	holdingitems += O
-	src.updateUsrDialog()
-	return 0
+	SSnano.update_uis(src)
+	return FALSE
 
 /obj/machinery/reagentgrinder/interface_interact(mob/user)
-	interact(user)
+	ui_interact(user)
 	return TRUE
 
 /obj/machinery/reagentgrinder/DefaultTopicState()
 	return GLOB.physical_state
 
-/obj/machinery/reagentgrinder/interact(mob/user as mob) // The microwave Menu
+/obj/machinery/reagentgrinder/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
 	if(inoperable())
-		return
+		to_chat(user, SPAN_WARNING("\The [src] is too broken to use."))
+		return FALSE
```

Comment:
I believe this check is antiquated and will never be false if this line is reached. If you want custom messages, you'd need to put them in `CouldNotUseTopic` or some similar proc.